### ISSUE TITLE
hashable classes from make_class

### ISSUE THREAD

**Issue by danodonovan** [python-attrs/attrs#234]
I've just stumbled across something I think might be missing from the ChangeLog;
```
>>> attr.__version__
'16.3.0'
>>> t = attr.make_class('test', ['item'])
>>> set([t(i) for i in range(4)])
{test(item=2), test(item=0), test(item=3), test(item=1)}
```
vs. 
```
>>> attr.__version__
'17.2.0'
>>> t = attr.make_class('test', ['item'])
>>> set([t(i) for i in range(4)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'test'
```

ie `attr.make_class` will return a hashable class in `16.3.0` but not in `17.2.0` (the `type(t.__hash__)` function is  `<class 'NoneType'>` in `17.2.0`). 

Only mentioning this as it was unexpected (to me at least!). Thanks for the excellent module - I never seem to write python code without it now.

**Comment by danodonovan** [comment]
Just noting that the for `17.2.0` you can just do;

    t = attr.make_class('test', ['item'], hash=True)

**Comment by hynek** [comment]
Well, the change landed in 17.1 and there’s a long paragraph under “Backward-incompatible changes”…I guess that’s unfortunate if you skip that version but I wouldn’t know how to alleviate that…

**Comment by danodonovan** [comment]
Ah yes. Sorry for the noise.
